lib/workers/global/autodiscover.ts

```diff
@@ -30,7 +30,16 @@ export async function autodiscoverRepositories(
     return config;
   }
   if (config.autodiscoverFilter) {
-    discovered = discovered.filter(minimatch.filter(config.autodiscoverFilter));
+    const matched = new Set<string>();
+
+    for (const filter of config.autodiscoverFilter) {
+      const res = minimatch.match(discovered, filter);
+      if (res.length > 0) {
+        res.forEach((e) => matched.add(e));
+      }
```

Comment:
```suggestion
      res.forEach((e) => matched.add(e));
```

I think we don't need the condition

Reply:
Done